api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go

```diff
@@ -198,8 +215,11 @@ func (m *CatalogConfigProvider) traceValue(v interface{}, localcontext interface
 		return val, nil
 	}
 }
+
+// TODO: IConfigProvider interface methods shoukd be enhanced to accept scope as a parameter
+// so we can get rid of getCatalogInDefaultNamespace
 func (m *CatalogConfigProvider) Set(object string, field string, value interface{}) error {
-	catalog, err := utils.GetCatalog(context.TODO(), m.Config.BaseUrl, object, m.Config.User, m.Config.Password)
+	catalog, err := m.getCatalogInDefaultNamespace(context.TODO(), m.Config.BaseUrl, object, m.Config.User, m.Config.Password)
```

Comment:
Looks like we only support catalog set in default namespace?

Reply:
Correct. The functionality of the Set method remains the same (work in the default namespace only). It's not the purpose of this PR to update it.